openapi/schemas/processes-core/statusInfo.yaml

```diff
@@ -1,42 +1,47 @@
-type: object
-required:
-   - id
-   - status
-   - type
-properties:
-   id:
-      type: string
-   processID:
-      type: string
-   type:
-      type: string
-      example:
-        - process
-        - wps
-        - openeo
-   status:
-      $ref: "statusCode.yaml"
-   message:
-      type: string
-   exception:
-      $ref: "../common-core/exception.yaml"
-   created:
-      type: string
-      format: date-time
-   started:
-      type: string
-      format: date-time
-   finished:
-      type: string
-      format: date-time
-   updated:
-      type: string
-      format: date-time
-   progress:
-      type: integer
-      minimum: 0
-      maximum: 100
-   links:
-      type: array
-      items:
-         $ref: "../common-core/link.yaml"
+allOf:
+  - $ref: descriptionType.yaml
+  - type: object
+    required:
+       - id
+       - status
+       - created
+       - process
+       - type
+    properties:
+       id:
+          type: string
+       process:
+          $ref: process.yaml
+       type:
```

Comment:
I've updated the description to indicate that if "type" is "openeo" then the "process" property MUST be present and its value shall be a process graph as per the OpenEO API definition.  Otherwise if can be an OAProc process description, a WPS process description, a link to either of these or a link to the request that created the job (e.g. an async coverages request).

Reply:
Do we want this condition to be more explicit?

Since we have the `allOf`, we can make use of it.

```yaml
allOf:
  - $ref: descriptionType.yaml
  - type: object
    # ... all other fields ...
  - if:
      properties:
        type: 
          const: openeo  # or 'enum' equivalent if prefered
    then:
      required:
        - id
        - type
        - status
        - process
      properties:
        process:
          $ref: "<uri-to-openeo-process-graph>"
```

Reply:
I'm wondering whether OGC API - Processes should just define the framework and not have direct dependencies to openEO due to the issue that there are multiple versions of the openEO API...

As long as openEO follows your framework, I guess it's fine, isn't it?

Reply:
I'm expecting that anyone that wants to do strong schema validation would actually use the reference openEO schemas rather than depend on OGC API, so it's fine if we don't define field requirements of openEO explicitly here. 

However, if we do so, I don't see much point to have the long description either for specific frameworks. Saying in the description that `process` is required when `type: openeo` doesn't actually enforce or validate anything.

Reply:
Yes, I think the spec should not include specifics about openEO.